### PR TITLE
Billing Page iteration one

### PR DESCRIPTION
**Pull request by MrFlashAccount**
#### Tl;dr 
Closes: enso-org/cloud-v2#1158
This PR adds a redirect to the stripe page where the user can manage his subscription

<details><summary>Demo Presentation</summary>
<p>

https://github.com/enso-org/enso/assets/61194245/360fdc7e-46ff-49fa-9936-e5c61fe6f917

</p>
</details> 

---

#### Context:
Our first iteration was to add our billing page but after a few iterations, we decided to postpone it in favor of more important features.

#### This Change:
1. creates a link for a private user session
2. redirect the user to that page (open in a new tab) when the user clicks on the billing tab

#### Test Plan:
Go over how you plan to test it. Your test plan should be more thorough the riskier the change is. For major changes, I like to describe how I E2E tested it and will monitor the rollout.

---

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [ ] Unit tests have been written where possible.

### CODE REVIEW

File: app/ide-desktop/lib/assets/credit_card.svg
Comment (somebody1234) [inline app/ide-desktop/lib/assets/credit_card.svg:1]: nit: missing rounded corners? also possibly darken the background of at least one of the cards to give it more visual weight to roughly match the other icons. also i suppose we should be making sure the lines are 2px wide, not 1px

**Review by somebody1234 — APPROVED** [review]
cr passed

File: app/ide-desktop/lib/assets/credit_card.svg
Comment (somebody1234) [reply app/ide-desktop/lib/assets/credit_card.svg:1]: (optional, of course. e.g. if this was taken from some design system somewhere we can always remake it later.)


oh also worth noting that when we have overlapping parts of the icon we usually either mask out an outline around the thing in front to make it stand out more  - or more rarely lower the opacity of one of the things

Comment (MrFlashAccount) [reply app/ide-desktop/lib/assets/credit_card.svg:1]: What do you mean by rounded corners? I see no difference with the icon from the figma: https://www.figma.com/design/MXE4HsnOgBwI49aXg0byIg/ide2-v124?node-id=21102-39250&t=aNh7aJiNbzzj93Jx-4

In general, we have huge issues with the current icons (their perceptional sizes aren't balanced), eg:

CleanShot 2024-07-10 at 16 55 40@2x

vs 

CleanShot 2024-07-10 at 16 56 00@2x

Compare to: https://www.figma.com/design/4vAG8Oino0kNO5fhKDiCG6/central%2Bicon%2Bsystem%2Bv1.17?node-id=5934-15120&t=ckFHoczoKZ415AzL-4

Comment (somebody1234) [reply app/ide-desktop/lib/assets/credit_card.svg:1]: oh my bad, the github preview is just being really weird

**Review by farmaazon — APPROVED** [review]
Approving changes in common lib.

**Comment by MrFlashAccount** [comment]
The related lambda is merged. This PR is ready for the QA. @somebody1234 , @PabloBuchu

File: app/dashboard/src/providers/BackendProvider.tsx [hunk 1]
```diff
@@ -61,9 +64,10 @@ export function useRemoteBackend() {
...
 // ==============================
 
-/** Get the Remote Backend.
- * @throws {Error} when no Remote Backend exists. This should only happen if the user is not logged
- * in. */
+/**
+ * Get the Remote Backend.
+ * @throws {Error} when no Remote Backend exists. This should never happen.
+ */
 export function useRemoteBackendStrict() {
```
Comment (somebody1234) [inline app/dashboard/src/providers/BackendProvider.tsx:71]: i guess this means we should gradually phase out `useRemoteBackendStrict` (or just remove all usages immediately here)

**Review by somebody1234 — APPROVED** [review]
CR :white_check_mark:

**Review by PabloBuchu — APPROVED** [review]

**Comment by PabloBuchu** [comment]
@MrFlashAccount any way to show team plan when I hit `update plan button`? currently I can only buy multiple solo plans which doesnt make much sense. If not maybe we should hide it and let user upgrade via our site

**Comment by MrFlashAccount** [comment]
@PabloBuchu This can be configured in the Stripe settings: https://docs.stripe.com/products-prices/pricing-models#flat-rate

<blockquote><div><strong><a href="https://docs.stripe.com/products-prices/pricing-models#flat-rate">Recurring pricing models</a></strong></div><div>Learn about common pricing models and how to create them.</div></blockquote>
